README
ibus-pinyin
It is a Chinese Pinyin input method for IBus.


debian/changelog
ibus-pinyin (1.3.3-1) unstable; urgency=low

  * New upstream release.

 -- Peng Huang <shawn.p.huang@gmail.com>  Fri, 16 Apr 2010 12:23:00 +0800

ibus-pinyin (1.3.1-1) unstable; urgency=low

  * New upstream release.

 -- LI Daobing <lidaobing@debian.org>  Sat, 10 Apr 2010 15:32:18 +0800

ibus-pinyin (1.3.0-1) unstable; urgency=low

  * New upstream release.
  * debian/source/format: 3.0.

 -- LI Daobing <lidaobing@debian.org>  Sat, 03 Apr 2010 21:22:28 +0800

ibus-pinyin (1.2.99.20100318-1) experimental; urgency=low

  * New upstream release.
  * debian/control: bump standards version to 3.8.4.
  * debian/control: switch to ibus 1.3.
  * debian/control: build depends on libsigc++-2.0-dev. 
  * debian/control: fix debhelper-but-no-misc-depends.
  * debian/rules: remove config.log 

 -- LI Daobing <lidaobing@debian.org>  Sun, 21 Mar 2010 19:22:14 +0800

ibus-pinyin (1.2.99.20091211-1) unstable; urgency=low

  * New upstream release.
  * debian/control: add build depends on intltool 

 -- LI Daobing <lidaobing@debian.org>  Sat, 12 Dec 2009 22:43:01 +0800

ibus-pinyin (1.2.99.20091009-1) unstable; urgency=low

  * new upstream release.
  * debian/control: rename package ibus-pinyin-db-google to
    ibus-pinyin-db-android.
  * debian/rules: updated. 

 -- LI Daobing <lidaobing@debian.org>  Sun, 11 Oct 2009 12:58:12 +0800

ibus-pinyin (1.2.99.20091007-1) unstable; urgency=low

  * new upstream release.
  * debian/control: remove package ibus-pinyin-data, add packages
    ibus-pinyin-db-open-phrase and ibus-pinyin-db-google.
  * debian/rules: updated.

 -- LI Daobing <lidaobing@debian.org>  Thu, 08 Oct 2009 12:39:25 +0800

ibus-pinyin (1.2.99.20090930-1) unstable; urgency=low

  * new upstream release.
  * debian/control: update build depends.

 -- LI Daobing <lidaobing@debian.org>  Wed, 30 Sep 2009 22:44:05 +0800

ibus-pinyin (1.2.99.20090929-1) unstable; urgency=low

  * new upstream release.
  * debian/watch: updated. 
  * debian/control: 
    - build depends on pinyin-database and lsb-release.
    - add ibus-pinyin-data package.
  * debian/rules: updated. 
  * debian/README.Debian: no longer needed. 

 -- LI Daobing <lidaobing@debian.org>  Tue, 29 Sep 2009 22:33:06 +0800

ibus-pinyin (1.2.99.20090923+1-1) unstable; urgency=low

  * new upstream release.
  * debian/rules: update build script. 
  * debian/control: 
    - change arch from all to any. 
    - remove Build-Depends-Indep, update Build-Depends
    - update Depends 

 -- LI Daobing <lidaobing@debian.org>  Wed, 23 Sep 2009 20:12:19 +0800

ibus-pinyin (1.2.0.20090915+1-1) unstable; urgency=low

  * new upstream release.
  * debian/control:
    - bump standards version to 3.8.3.
    - update Vcs-* fields. 

 -- LI Daobing <lidaobing@debian.org>  Tue, 15 Sep 2009 20:09:45 +0800

ibus-pinyin (1.2.0.20090617+1-1) unstable; urgency=low

  * new upstream release.
  * debian/control:
    - depends: ibus >= 1.2
    - bump standards version to 3.8.2

 -- LI Daobing <lidaobing@debian.org>  Sat, 20 Jun 2009 23:40:41 +0800

ibus-pinyin (1.1.0.20090612+1-1) unstable; urgency=low

  * new upstream release.

 -- LI Daobing <lidaobing@debian.org>  Sat, 13 Jun 2009 15:37:23 +0800

ibus-pinyin (1.1.0.20090303+1-0ubuntu1) karmic; urgency=low

  * initial release (LP: #318806)

 -- LI Daobing <lidaobing@debian.org>  Fri, 01 May 2009 16:44:56 +0800


src/Types.h
/* vim:set et sts=4: */
#ifndef __PY_TYPE_H_
#define __PY_TYPE_H_

#include <glib.h>

namespace PY {

#define PINYIN_ID_VOID  (-1)
#define PINYIN_ID_ZERO  (0)
#define PINYIN_ID_B     (1)
#define PINYIN_ID_C     (2)
#define PINYIN_ID_CH    (3)
#define PINYIN_ID_D     (4)
#define PINYIN_ID_F     (5)
#define PINYIN_ID_G     (6)
#define PINYIN_ID_H     (7)
#define PINYIN_ID_J     (8)
#define PINYIN_ID_K     (9)
#define PINYIN_ID_L     (10)
#define PINYIN_ID_M     (11)
#define PINYIN_ID_N     (12)
#define PINYIN_ID_P     (13)
#define PINYIN_ID_Q     (14)
#define PINYIN_ID_R     (15)
#define PINYIN_ID_S     (16)
#define PINYIN_ID_SH    (17)
#define PINYIN_ID_T     (18)
#define PINYIN_ID_W     (19)
#define PINYIN_ID_X     (20)
#define PINYIN_ID_Y     (21)
#define PINYIN_ID_Z     (22)
#define PINYIN_ID_ZH    (23)
#define PINYIN_ID_A     (24)
#define PINYIN_ID_AI    (25)
#define PINYIN_ID_AN    (26)
#define PINYIN_ID_ANG   (27)
#define PINYIN_ID_AO    (28)
#define PINYIN_ID_E     (29)
#define PINYIN_ID_EI    (30)
#define PINYIN_ID_EN    (31)
#define PINYIN_ID_ENG   (32)
#define PINYIN_ID_ER    (33)
#define PINYIN_ID_I     (34)
#define PINYIN_ID_IA    (35)
#define PINYIN_ID_IAN   (36)
#define PINYIN_ID_IANG  (37)
#define PINYIN_ID_IAO   (38)
#define PINYIN_ID_IE    (39)
#define PINYIN_ID_IN    (40)
#define PINYIN_ID_ING   (41)
#define PINYIN_ID_IONG  (42)
#define PINYIN_ID_IU    (43)
#define PINYIN_ID_O     (44)
#define PINYIN_ID_ONG   (45)
#define PINYIN_ID_OU    (46)
#define PINYIN_ID_U     (47)
#define PINYIN_ID_UA    (48)
#define PINYIN_ID_UAI   (49)
#define PINYIN_ID_UAN   (50)
#define PINYIN_ID_UANG  (51)
#define PINYIN_ID_UE    (52)
#define PINYIN_ID_VE    PINYIN_ID_UE
#define PINYIN_ID_UI    (53)
#define PINYIN_ID_UN    (54)
#define PINYIN_ID_UO    (55)
#define PINYIN_ID_V     (56)
#define PINYIN_ID_NG    PINYIN_ID_VOID

#define PINYIN_INCOMPLETE_PINYIN    (1U << 0)

#define PINYIN_CORRECT_GN_TO_NG     (1U << 1)
#define PINYIN_CORRECT_MG_TO_NG     (1U << 2)
#define PINYIN_CORRECT_IOU_TO_IU    (1U << 3)
#define PINYIN_CORRECT_UEI_TO_UI    (1U << 4)
#define PINYIN_CORRECT_UEN_TO_UN    (1U << 5)
#define PINYIN_CORRECT_UE_TO_VE     (1U << 6)
#define PINYIN_CORRECT_VE_TO_UE     (1U << 7)
#define PINYIN_CORRECT_ALL          (0x000000fe)

#define PINYIN_FUZZY_C_CH           (1U << 8)
#define PINYIN_FUZZY_CH_C           (1U << 9)
#define PINYIN_FUZZY_Z_ZH           (1U << 10)
#define PINYIN_FUZZY_ZH_Z           (1U << 11)
#define PINYIN_FUZZY_S_SH           (1U << 12)
#define PINYIN_FUZZY_SH_S           (1U << 13)
#define PINYIN_FUZZY_L_N            (1U << 14)
#define PINYIN_FUZZY_N_L            (1U << 15)
#define PINYIN_FUZZY_F_H            (1U << 16)
#define PINYIN_FUZZY_H_F            (1U << 17)
#define PINYIN_FUZZY_L_R            (1U << 18)
#define PINYIN_FUZZY_R_L            (1U << 19)
#define PINYIN_FUZZY_K_G            (1U << 20)
#define PINYIN_FUZZY_G_K            (1U << 21)

#define PINYIN_FUZZY_AN_ANG         (1U << 22)
#define PINYIN_FUZZY_ANG_AN         (1U << 23)
#define PINYIN_FUZZY_EN_ENG         (1U << 24)
#define PINYIN_FUZZY_ENG_EN         (1U << 25)
#define PINYIN_FUZZY_IN_ING         (1U << 26)
#define PINYIN_FUZZY_ING_IN         (1U << 27)
#define PINYIN_FUZZY_IAN_IANG       (1U << 28)
#define PINYIN_FUZZY_IANG_IAN       (1U << 29)
#define PINYIN_FUZZY_UAN_UANG       (1U << 30)
#define PINYIN_FUZZY_UANG_UAN       (1U << 31)
#define PINYIN_FUZZY_ALL            (0xffffff00)

struct Pinyin {
    const gchar *text;
    const gchar *sheng;
    const gchar *yun;
    const gchar  sheng_id;
    const gchar  yun_id;
    const gchar  fsheng_id;
    const gchar  fyun_id;
    const gchar  fsheng_id_2;
    const gchar  fyun_id_2;
    const guint  len;
    const guint  flags;
};

#define MAX_UTF8_LEN 6
#define MAX_PHRASE_LEN 16

};

#endif
